crates/network-scanner/src/mdns.rs

```diff
@@ -110,3 +115,35 @@ pub fn mdns_query_scan(
         });
     Ok(result_receiver)
 }
+
+fn parse_fullname(fullname: &str) -> Option<(String, Option<Protocol>)> {
+    let mut iter = fullname.split('.');
+    let device_name = iter.next()?;
+    let mut service_type = String::new();
+    while let Some(part) = iter.next() {
+        if part.starts_with("_") {
+            service_type.push_str(part);
+            service_type.push('.');
+        }
+    }
+    // remove the trailing dot
+    service_type.pop()?;
+
+    let protocol = match service_type.as_str() {
+        "_http._tcp" => Some(Protocol::Http),
+        "_https._tcp" => Some(Protocol::Https),
+        "_ssh._tcp" => Some(Protocol::Ssh),
+        "_sftp._tcp" => Some(Protocol::Sftp),
+        "_scp._tcp" => Some(Protocol::Scp),
+        "_telnet._tcp" => Some(Protocol::Telnet),
+        "_ldap._tcp" => Some(Protocol::Ldap),
+        "_ldaps._tcp" => Some(Protocol::Ldaps),
+        // https://jonathanmumm.com/tech-it/mdns-bonjour-bible-common-service-strings-for-various-vendors/
+        // OSX Screen Sharing
+        "_rfb._tcp" => Some(Protocol::Ard),
+        // Note: RDP, VNC, Wayk, and SSH-based PowerShell (SshPwsh), as well as tunneling (Tunnel), are not standardized service types
```

Comment:
nitpick: JMUX (Devolutions Gateway Tunnel) is not worth mentioning here.

```suggestion
        // Note: RDP, VNC, Wayk, and SSH-based PowerShell (SshPwsh) are not standardized service types
```